### PR TITLE
select_datasets added to pmlb.py

### LINKED ISSUE #3: Extend Python wrapper
We should extend the Python wrapper to allow the user to query the repository for data sets that meet a certain criteria.

For example, the user could query for "all data sets that have >=1000 records" or "all data sets that are binary classification problems."

### PR DESCRIPTION
**Pull request by praneelc**
Closes #3, #87.

### CODE REVIEW

File: pmlb/pmlb.py [hunk 1]
```diff
@@ -256,4 +256,57 @@ def get_reviewed_datasets(dataset_names, local_cache_dir = 'datasets/'):
         if header != '# Reviewed by [your name here]\n':
             reviewed_datasets.append(dataset_name)
             
-    return sorted(reviewed_datasets)
\ No newline at end of file
+    return sorted(reviewed_datasets)
+
+def filter_datasets(obs_min = None, obs_max = None, feat_min = None, feat_max = None, class_min = None, class_max = None, endpt = None, max_imbalance = None, task = None):
```
Comment (trangdata) [inline pmlb/pmlb.py]: I think we decided on `select_datasets()`.
```suggestion
def select_datasets(obs_min = None, obs_max = None, feat_min = None, feat_max = None, class_min = None, class_max = None, endpt = None, max_imbalance = None, task = None):
```

File: pmlb/pmlb.py [hunk 2]
```diff
@@ -256,4 +256,57 @@ def get_reviewed_datasets(dataset_names, local_cache_dir = 'datasets/'):
...
+         Whether the dataset is suited for classification or regression problems
+     Returns
+     ----------
+     list (str): 
+         list of names of datasets within filters. Will return an empty list if no datasets match.
+         
+     
+     """
+
+     tempdf = pd.read_csv('https://raw.githubusercontent.com/EpistasisLab/penn-ml-benchmarks/master/datasets/all_summary_stats.csv')
```
Comment (trangdata) [inline pmlb/pmlb.py]: Maybe use relative path here

```suggestion
    path = pathlib.Path(__file__).parent / "all_summary_stats.tsv"
    tempdf = pd.read_csv(path, sep = '\t')
```

File: pmlb/pmlb.py [hunk 3]
```diff
@@ -256,4 +256,57 @@ def get_reviewed_datasets(dataset_names, local_cache_dir = 'datasets/'):
         if header != '# Reviewed by [your name here]\n':
             reviewed_datasets.append(dataset_name)
             
-    return sorted(reviewed_datasets)
\ No newline at end of file
+    return sorted(reviewed_datasets)
+
+def filter_datasets(obs_min = None, obs_max = None, feat_min = None, feat_max = None, class_min = None, class_max = None, endpt = None, max_imbalance = None, task = None):
+     """Filters existing datasets by given parameters, and returns a list of their names.
```
Comment (trangdata) [inline pmlb/pmlb.py]: I counted 5 spaces of indentation here. We should probably check to make sure it's the 4-space indent.

**Review by trangdata — COMMENTED** [review]
Looks good, but I had some comments. We should also add this function to export in [`__init.py__`](https://github.com/EpistasisLab/penn-ml-benchmarks/blob/master/pmlb/__init__.py).

**Comment by praneelc** [comment]
[Colab Notebook with usage example](https://colab.research.google.com/drive/1g74vHGDQYrUEYY35SJVjZuoXkw9OU78O#scrollTo=PzAInO9ndtDP)

**Pull request merged**